src/Storages/RabbitMQ/RabbitMQSource.cpp

```diff
@@ -120,10 +120,20 @@ Chunk RabbitMQSource::generateImpl()
     {
         auto timeout = std::chrono::milliseconds(context->getSettingsRef().rabbitmq_max_wait_ms.totalMilliseconds());
         consumer = storage.popConsumer(timeout);
+
+        if (consumer->needChannelUpdate())
+        {
```

Comment:
I'm not sure if that is practical, yet `popConsumer` is designed to return `nullptr` if the waiting timeout is exceeded and no consumer has become available by that time:
https://github.com/ClickHouse/ClickHouse/blob/master/src/Storages/RabbitMQ/StorageRabbitMQ.cpp#L901

Should be accounted for.